File: lab7/DictionaryTest.java
// -----------------------------------------------------------------------------
// Yongshi Sun ID:1619410
// A test client for the Dictionary ADT
// ----------------------------------------------------------------------------

public class DictionaryTest {

    public static void main(String[] args) {
        Dictionary A = new Dictionary();
        Dictionary B = new Dictionary();

        System.out.println("isEmpty: " + A.isEmpty());
        B.insert("1", "b");
        B.insert("2", "a");
        B.insert("3", "g");
        System.out.println("isEmpty: " + B.isEmpty());
        System.out.println(B);
        System.out.println("Size: " + B.size());
        String v = B.lookup("3");
        System.out.println("key=1 " + (v == null ? "not found" : ("value=" + v)));
        B.delete("1");
        System.out.println(B);
        System.out.println("Size: " + B.size());
        B.insert("4", "c");
        System.out.println(B);
        B.makeEmpty();
        System.out.println("isEmpty: " + B.isEmpty());

    }
}
 
 
 

File: pa5/DictionaryTest.c
//-----------------------------------------------------------------------------
// DictionaryTest.c
// Test client for the Dictionary ADT
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
    Dictionary A = newDictionary();
    char* k;
    char* v;

    printf("%s\n", (isEmpty(A)?"true":"false"));
    insert(A,"one", "hi");
    insert(A,"two", "bro");
    //insert(A,"two", "bro");
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", (size(A)));
    printDictionary(stdout, A);
    
    delete(A, "two" );
    printf("%d\n", (size(A)));

    printDictionary(stdout, A);
    k = "one";
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);

    makeEmpty(A);
    printf("%s\n", (isEmpty(A)?"true":"false")); 


}